### ISSUE TITLE
research ability to grow ceph backed storage similar to s3 (see use case)

### ISSUE THREAD

**Issue by msdisme** [CCI-MOC/MOC-issues#113]
use case: if we want to save metrics, which will grow over time, we will need the ability to grow amount of storage on NESE.   

This is the same issue anyone with growing data sets will experience.

**Comment by msdisme** [comment]
@computate @schwesig for 2.0 should we consider either offsite (eg. wasabi, Backblaze Business) or storge to tape?  Is this something OSAC or MOC 2.0 should include as a validated pattern?  ( @hpdempsey @rstahl-rh )

**Comment by hpdempsey** [comment]
No one has asked for offsite storage AFAIK.  We've had some takers for offsite tape storage though.  I don't really think we'd want to cover that as a validated pattern because it is a ton of work to verify for a very small potential user base, and can be done as a separate step from the rest of the pattern.

We will of course need to figure out how to do an offsite Data Commons, but I don't think we're trying to tackle that yet in MOC 2.0.  Certainly not in the consultant's current version.

**Comment by hpdempsey** [comment]
I should say, no one except our offsite telemetry storage.  I don't think we've discussed how much telemetry to save for MOC 2.0, so we would need to clarify that before knowing whether we'd need offsite or tape backup.

**Comment by schwesig** [comment]
Offsite backup was essentially the last resort in MOC 1.0 to achieve any form of long-term retention. Because the Ceph cluster frequently exceeded 70% utilization, performance degraded significantly, making it impractical to keep growing datasets online. When the pause was announced, we had to quickly develop a backup strategy (see
- CCI-MOC/ops-issues#1788
).

For a properly planned MOC 2.0 deployment, I would expect this use case to be handled by the onsite infrastructure without requiring an emergency offsite backup solution.

I've already discussed the current storage requirements with the consultants. At the moment we're using about 5 TB, and we've included additional buffer capacity in the planning.

**Comment by hpdempsey** [comment]
Yes, agree we shouldn't need offsite storage for the new design.  I think that that the HIPAA requirements might also prohibit us from sharing this kind of data with outside researchers in the NIST/HIPAA compliance rules.  If so,  we won't need to keep as much telemetry data anyway.  Will have to set the max retention for logs and for telemetry as part of the new deployment.

**Comment by hpdempsey** [comment]
I'm not sure NESE is compliant with NIST/HIPAA either.  I suspect not, since they obviously have not been keeping up on versions/patching.  So that would rule it out for any project with large datasets, not just Observability.
